man/openmp-utils.Rd

```diff
@@ -38,20 +38,121 @@
 
   \itemize{
     \item\file{between.c} - \code{\link{between}()}
+    
+    OpenMP is used here to parallelize:
+    \itemize{
+      \item The loops that check if each element of the vector provided is between the specified \code{lower} and \code{upper} bounds, for integer (\code{INTSXP}) and real (\code{REALSXP}) types
+      \item The checking and handling of undefined values (such as NaNs)
+    }
+    
+    Since this function is used to find rows where a column's value falls within a specific range, it benefits more from parallelization when the input data consists of a large number of rows.
+    
     \item\file{cj.c} - \code{\link{CJ}()}
+    
+    OpenMP is used here to parallelize:
+
+    \itemize{
+      \item The element assignment in vectors
+      \item The memory copying operations (blockwise replication of data using \code{memcpy})
+      \item The creation of all combinations of the input vectors over the cross-product space
+    }
+    
+    Given that the number of combinations increases exponentially as more columns are added, better speedup can be expected when dealing with a large number of columns.
+    
     \item\file{coalesce.c} - \code{\link{fcoalesce}()}
+    
+    OpenMP is used here to parallelize:
+    \itemize{
+      \item The operation that iterates over the rows to coalesce the data (which can be of type integer, real, or complex)
+      \item The replacement of NAs with non-NA values from subsequent vectors
+      \item The conditional checks within parallelized loops
+      }
+    
+    Significant speedup can be expected for more number of columns here, given that this function operates efficiently across multiple columns to find non-NA values.
+    
     \item\file{fifelse.c} - \code{\link{fifelse}()}
+    
+    For logical, integer, and real types, OpenMP is being used here to parallelize loops that perform conditional checks along with assignment operations over the elements of the supplied logical vector based on the condition (\code{test}) and values provided for the remaining arguments (\code{yes}, \code{no}, and \code{na}).
+    
+    Better speedup can be expected for more number of columns here as well, given that this function operates column-wise with independent vector operations.
+    
     \item\file{fread.c} - \code{\link{fread}()}
+    
+    OpenMP is used here to:
+
+    \itemize{
+      \item Avoid race conditions or concurrent writes to the output \code{data.table} by having atomic operations on the string data 
```

Comment:
I would also mention that we use openmp here for the chunked reading of the file

Reply:
Absolutely! Not sure why I initially thought that the parallel region was only dealing with the progress updates, but it makes complete sense now that I re-read the code. (For starters, I should have clearly noticed the [directive](https://github.com/Rdatatable/data.table/blob/f5a1e09328fef5f8a4c5e225bc5b75ca09700f76/src/fread.c#L2222) besides the [loop](https://github.com/Rdatatable/data.table/blob/f5a1e09328fef5f8a4c5e225bc5b75ca09700f76/src/fread.c#L2266) before at least)